dubbo-configcenter/dubbo-configcenter-nacos/src/main/java/org/apache/dubbo/configcenter/support/nacos/NacosDynamicConfiguration.java

```diff
@@ -113,7 +114,18 @@ private void setServerAddr(URL url, Properties properties) {
     private void setProperties(URL url, Properties properties) {
         putPropertyIfAbsent(url, properties, NAMESPACE);
         putPropertyIfAbsent(url, properties, NACOS_NAMING_LOG_NAME);
-        putPropertyIfAbsent(url, properties, ENDPOINT);
+
+        String endpoint = url.getParameter(ENDPOINT);
+        if (StringUtils.isNotEmpty(endpoint)) {
+            if (endpoint.contains(":")) {
+                int index = endpoint.lastIndexOf(":");
+                properties.put(ENDPOINT, endpoint.substring(0, index));
+                properties.put(ENDPOINT_PORT, endpoint.substring(index + 1));
```

Comment:
It is recommended to put the method of parsing host and port into NetUtils.java.